src/Services/QueuedJobService.php

```diff
@@ -708,7 +708,11 @@ public function runJob($jobId)
                     }
 
                     if (!$broken) {
-                        // Collect output as job messages as well as sending it to the screen
+                        // Inject real-time logger
+                        $queuedJobLogger = new QueuedJobLogger($job, $jobDescriptor);
```

Comment:
Please use Injector here to allow it to be extensible:

```suggestion
                        $queuedJobLogger = QueuedJobLogger::create($job, $jobDescriptor);
```

(Requires suggestion above)